website/Makefile

```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+build: content/.jekyll-cache
+	@bazel run //runsc -- help syscalls -format json -filename $(PWD)/syscalls.json
+	@docker build -t gvisor-website .
+	@docker run -v $$PWD:/work -w /work gcr.io/cloud-builders/npm ci
+	@$(MAKE) update
+.PHONY: build
+
+content/.jekyll-cache:
+	@mkdir -p $@ && chmod a+rw $@
+
+lint:
+	@docker run -v $$PWD:/work -w /work gcr.io/cloud-builders/npm run lint-md
+.PHONY: lint
+
+update: lint
+	@sudo mkdir -p _site && sudo chmod a+rw _site
```

Comment:
Using sudo to build the website locally has a smell to it and you'll end up with a bunch of local files owned by root etc.

In the current Makefile I used -u to set the user id for docker run and this can often work ok.
```-u="$(shell id -u):$(shell id -g)"```
https://github.com/google/gvisor-website/blob/master/Makefile#L52

It's a little clunky but if the docker image really needs the user to exist in /etc/passwd then you can add it at runtime. Could also be part of the CMD in the Dockerfile if we want to keep the Makefile simple.
```
groupadd --gid $(shell id -g) $(shell id -gn) && \
useradd --uid $(shell id -u) --gid $(shell id -g) -ms /bin/bash $(USER) && \
su $(USER) -c "..."
```
https://github.com/google/gvisor-website/blob/master/Makefile#L101-L103

This way of doing is much simpler than trying to use user namespace mappings w/ Docker.

Reply:
Sudo no longer required!